java/java.lsp.server/nbproject/org-netbeans-modules-java-lsp-server.sig

```diff
@@ -0,0 +1,175 @@
+meth public final java.io.PrintWriter getOut(org.netbeans.modules.java.lsp.server.ui.AbstractLspInputOutputProvider$LspIO)
+meth public final java.io.Reader getIn(org.netbeans.modules.java.lsp.server.ui.AbstractLspInputOutputProvider$LspIO)
+meth public final java.lang.String getIODescription(org.netbeans.modules.java.lsp.server.ui.AbstractLspInputOutputProvider$LspIO)
+meth public final java.lang.String getId()
+meth public final java.lang.Void getCurrentPosition(org.netbeans.modules.java.lsp.server.ui.AbstractLspInputOutputProvider$LspIO,java.io.PrintWriter)
+meth public final java.lang.Void startFold(org.netbeans.modules.java.lsp.server.ui.AbstractLspInputOutputProvider$LspIO,java.io.PrintWriter,boolean)
+meth public final org.netbeans.modules.java.lsp.server.ui.AbstractLspInputOutputProvider$LspIO getIO(java.lang.String,boolean,org.openide.util.Lookup)
+meth public final org.openide.util.Lookup getIOLookup(org.netbeans.modules.java.lsp.server.ui.AbstractLspInputOutputProvider$LspIO)
+meth public final void closeIO(org.netbeans.modules.java.lsp.server.ui.AbstractLspInputOutputProvider$LspIO)
+meth public final void endFold(org.netbeans.modules.java.lsp.server.ui.AbstractLspInputOutputProvider$LspIO,java.io.PrintWriter,java.lang.Void)
+meth public final void print(org.netbeans.modules.java.lsp.server.ui.AbstractLspInputOutputProvider$LspIO,java.io.PrintWriter,java.lang.String,org.netbeans.api.io.Hyperlink,org.netbeans.api.io.OutputColor,boolean)
+meth public final void resetIO(org.netbeans.modules.java.lsp.server.ui.AbstractLspInputOutputProvider$LspIO)
+meth public final void scrollTo(org.netbeans.modules.java.lsp.server.ui.AbstractLspInputOutputProvider$LspIO,java.io.PrintWriter,java.lang.Void)
+meth public final void setFoldExpanded(org.netbeans.modules.java.lsp.server.ui.AbstractLspInputOutputProvider$LspIO,java.io.PrintWriter,java.lang.Void,boolean)
+meth public final void setIODescription(org.netbeans.modules.java.lsp.server.ui.AbstractLspInputOutputProvider$LspIO,java.lang.String)
+meth public final void showIO(org.netbeans.modules.java.lsp.server.ui.AbstractLspInputOutputProvider$LspIO,java.util.Set<org.netbeans.api.io.ShowOperation>)
+supr java.lang.Object
+
+CLSS public final static org.netbeans.modules.java.lsp.server.ui.AbstractLspInputOutputProvider$LspIO
+ outer org.netbeans.modules.java.lsp.server.ui.AbstractLspInputOutputProvider
+supr java.lang.Object
+hfds ctx,err,in,lookup,name,out
+hcls LspWriter
+
+CLSS public abstract org.netbeans.modules.java.lsp.server.ui.AbstractLspStatusDisplayer
+cons protected init()
+meth public final java.lang.String getStatusText()
+meth public final org.openide.awt.StatusDisplayer$Message setStatusText(java.lang.String,int)
+meth public final void addChangeListener(javax.swing.event.ChangeListener)
+meth public final void removeChangeListener(javax.swing.event.ChangeListener)
+meth public final void setStatusText(java.lang.String)
+supr org.openide.awt.StatusDisplayer
+hfds text
+
```

Comment:
Yup, various `Abstract*` providers and `IOContext` and `UIContext` classes are exposed from `java/java.lsp.server` to `java/java.lsp.server/nbcode` suite which registers them as a service.

I decided to follow this `Abstract*` implementation followed by registration style, because it lowers the amount of API `java.lsp.server` needs to expose. For example now we don't have to export anything from the `lsp4j` Eclipse API. Its usage remains hidden inside the `Abstract*` classes.

Reply:
@JaroslavTulach shouldn't we make this API rather a Friend one ? It's rather immature you know.

Reply:
Don't worry. We'll deprecated when necessary. I don't want to promote usage of friend API myself.